hs_core/views/utils.py

```diff
@@ -687,8 +708,70 @@ def unzip_file(user, res_id, zip_with_rel_path, bool_remove_original):
         raise ValidationError("Unzipping of this file is not supported.")
 
     zip_fname = os.path.basename(zip_with_rel_path)
-    unzip_path = istorage.unzip(zip_with_full_path)
-    link_irods_folder_to_django(resource, istorage, unzip_path)
+    working_dir = os.path.dirname(zip_with_full_path)
+    unzip_path = None
+    try:
+
+        if overwrite:
+            # irods doesn't allow overwrite, so we have to check if a file exists, delete it and
+            # then write the new file. Aggregations are treated as single objects.  If one file is
+            # overwritten in an aggregation, the whole aggregation is deleted.
+
+            # unzip to a temporary folder
```

Comment:
@sblack-usu I am not sure about the comment ```irods doesn't allow overwrite```. With a ```-f``` flag, I think iRODS will allow overwrite of files with the same file name. Let me know if I missed anything on this.

Reply:
https://docs.irods.org/4.2.1/icommands/user/#imv

You're likely thinking of iput?